lib/reply.js

```diff
@@ -156,16 +156,17 @@ function pumpCallback (reply) {
 
 function handleError (reply, error, cb) {
   var statusCode = reply.res.statusCode
-  if (error == null) {
-    statusCode = statusCode || 500
-  } else if (error.status >= 400) {
-    statusCode = error.status
-  } else if (error.statusCode >= 400) {
-    statusCode = error.statusCode
-  } else {
-    statusCode = statusCode || 500
+
+  if (error) {
+    if (error.status >= 400) {
+      statusCode = error.status
+    } else if (error.statusCode >= 400) {
+      statusCode = error.statusCode
+    } else if (statusCode === 200) { // default status code
+      statusCode = 500
+    }
   }
-  if (statusCode < 400) statusCode = 500
```

Comment:
I’m -1 on removing this.

Reply:
I produce the same behaviour. As soon an error object is passed and no statusCode was changed we will change it to 500.

Reply:
https://github.com/fastify/fastify/pull/601/files#diff-d2fdf63c1d7f1d9f68270b6dfa8c930cR165